widgets/viewports/renderer.cpp

```diff
@@ -1903,6 +1890,129 @@ hash_list<nodeListElement *> ViewportBase::pickNodes(int centerX, int centerY, i
+
+        int shift{0};
+        if (selectionPass == RenderOptions::SelectionPass::NodeID24_48Bits) {
+            shift = 24;
+        } else if (selectionPass == RenderOptions::SelectionPass::NodeID48_64Bits) {
+            shift = 48;
+        }
+        const auto bits = static_cast<GLuint>(name >> shift);// extract 24 first, middle or 16 last bits of interest
+        color.setRed(static_cast<std::uint8_t>(bits));
+        color.setGreen(static_cast<std::uint8_t>(bits >> 8));
+        color.setBlue(static_cast<std::uint8_t>(bits >> 16));
+        color.setAlpha(255);
+
+        return arrayFromQColor(color);
+    };
+
+    auto addNode = [arrayFromQColor, getPickingColor, options](const nodeListElement & node) {
+        auto color = state->viewer->getNodeColor(node);
+
+        if (node.selected && options.highlightSelection) {// highlight selected nodes
+            auto selectedNodeColor = QColor(Qt::green);
+//            selectedNodeColor.setAlphaF(0.5f);// results in half-transparent nodes in low mode
+            color = selectedNodeColor;
+        }
+
+        const auto isoPos = Dataset::current().scale.componentMul(node.position);
+
+        state->viewerState->colorPickingBuffer24.emplace_back(getPickingColor(node, RenderOptions::SelectionPass::NodeID0_24Bits));
+        state->viewerState->colorPickingBuffer48.emplace_back(getPickingColor(node, RenderOptions::SelectionPass::NodeID24_48Bits));
+        state->viewerState->colorPickingBuffer64.emplace_back(getPickingColor(node, RenderOptions::SelectionPass::NodeID48_64Bits));
+        state->viewerState->pointVertBuffer.emplace_back(isoPos, arrayFromQColor(color));
+    };
+
+    const auto * activeTree = state->skeletonState->activeTree;
+    const auto * activeNode = state->skeletonState->activeNode;
+    const auto * activeSynapse = (activeNode && activeNode->isSynapticNode) ? activeNode->correspondingSynapse :
+                                 (activeTree && activeTree->isSynapticCleft) ? activeTree->correspondingSynapse :
+                                                                               nullptr;
+    const bool synapseBuilding = state->skeletonState->synapseState != Synapse::State::PreSynapse;
+    const bool onlySelected = state->viewerState->skeletonDisplay.testFlag(TreeDisplay::OnlySelected);
+
+    for (auto & currentTree : Skeletonizer::singleton().skeletonState.trees) {
+        // focus on synapses, darken rest of skeleton
+        const bool darken = (synapseBuilding && currentTree.correspondingSynapse != &state->skeletonState->temporarySynapse)
+                || (activeSynapse && activeSynapse->getCleft() != &currentTree);
+        const bool hideSynapses = !darken && !synapseBuilding && !activeSynapse && currentTree.isSynapticCleft;
+        const bool selectionFilter = onlySelected && !currentTree.selected;
+        if (selectionFilter || (!currentTree.render && (!currentTree.isSynapticCleft || (currentTree.isSynapticCleft && hideSynapses)) )) {
```

Comment:
this `if` should be due to a function call which is used in `renderSkeleton` too